app/Domain/Slack/Traits/FindsUserBySlackId.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Toby\Domain\Slack\Traits;
+
+use Illuminate\Support\Str;
+use Toby\Domain\Slack\Exceptions\UserNotFoundException;
+use Toby\Eloquent\Models\User;
+
+trait FindsUserBySlackId
+{
+    protected function findUserBySlackId(string $slackId): ?User
+    {
+        $id = $this->prepareSlackIdFromString($slackId);
+
+        /** @var User $user */
```

Comment:
Is it required here? https://github.com/laravel/framework/blob/9.x/src/Illuminate/Collections/Collection.php#L387

Reply:
Yes.
`Return value is expected to be 'null|\Toby\Eloquent\Models\User', '\Illuminate\Database\Eloquent\Builder|\Illuminate\Database\Eloquent\Model|object' returned `